src/map_io/map_buildingdata_packet.cc

```diff
@@ -782,14 +781,16 @@ void MapBuildingdataPacket::read_trainingsite(TrainingSite& trainingsite,
                                               const TribesLegacyLookupTable& tribes_lookup_table) {
 	try {
 		uint16_t const packet_version = fr.unsigned_16();
-		if (packet_version == kCurrentPacketVersionTrainingsite) {
+		// TODO(unknown): remove support for previous packet after release 21, to keep code simple.
+		if (packet_version == kCurrentPacketVersionTrainingsite || packet_version == kCurrentPacketVersionTrainingsite -1) {
```

Comment:
```suggestion
		if (packet_version <= kCurrentPacketVersionTrainingsite && packet_version >= 5) {
```

Reply:
I think that the original is more intuitive when one removes the support for old versions, but both work similarly well. No strong feelings.

Reply:
This is just for consistency because we hardcode the legacy version numbers everywhere else. And when the version is increased next, using relative versions could have a nasty side effect if this is overlooked.